src/generic.rs

```diff
@@ -686,10 +694,10 @@ impl<T> Cancellable<T> {
     }
 }
 
-impl<F, T> Future for Cancellable<F>
+impl<'py, F, T> Future for Cancellable<F>
 where
     F: Future<Output = PyResult<T>>,
-    T: IntoPy<PyObject>,
+    T: IntoPyObject<'py>,
```

Comment:
This one if _probably_ the only one where you need the higher ranked bound, because there is no input to tie it to.